components/experimental/src/measure/measureunit.rs

```diff
@@ -77,9 +78,9 @@ impl MeasureUnit {
     /// ```
     pub fn generate_short_representation(&self) -> String {
         // Convert the constant to scientific notation if it is a power of 10 with more than 3 trailing zeros
-        fn append_power_of_10_to_scientific(n: u64, append_to: &mut String) {
+        fn append_power_of_10_to_scientific(n: u64, buff: &mut String) {
             if n < 1000 {
-                append_to.push_str(&n.to_string());
+                write!(buff, "{}", n).unwrap();
```

Comment:
this is infallible even though the API is fallible, which is why my suggestion was
```suggestion
                let _infallible = write!(buff, "{}", n);
```
this way no panicking code gets linked, and Rust doesn't complain about not using a `Result`.